packages/ember-glimmer/lib/utils/references.js

```diff
@@ -78,7 +78,7 @@ class PropertyReference extends CachedReference { // jshint ignore:line
 
     _parentObjectTag.update(tagFor(parentValue));
 
-    if (parentValue && typeof parentValue === 'object') {
+    if (parentValue && typeof parentValue === 'object' && typeof _propertyKey === 'string') {
```

Comment:
See https://github.com/emberjs/ember.js/issues/13296 and https://github.com/emberjs/ember.js/pull/13311.  We should support `{{get foo 1}}` (iow, not just string keys).